rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -341,72 +362,225 @@ def _convert_attribute_tokens_to_texts(
+    ) -> None:
+        """Update the existing vocabulary of the vectorizer with new unseen words.
 
-        combined_cleaned_texts = []
-        for attribute in self._attributes:
-            combined_cleaned_texts += attribute_texts[attribute]
+        These unseen words should only occupy the empty buffer slots.
 
-        try:
-            self.vectorizers[TEXT].fit(combined_cleaned_texts)
-        except ValueError:
+        Args:
+            attribute: Message attribute for which vocabulary should be updated.
+            vocabulary: Set of words to expand the vocabulary with if they are unseen.
+
+        """
+        existing_vocabulary: Dict[Text, int] = self.vectorizers[attribute].vocabulary
+        available_empty_index = self._get_starting_empty_index(existing_vocabulary)
+        if len(vocabulary) > len(existing_vocabulary):
             logger.warning(
-                "Unable to train a shared CountVectorizer. "
-                "Leaving an untrained CountVectorizer"
+                f"New data contains vocabulary of size {len(vocabulary)} for attribute {attribute} "
+                f"which is larger than the maximum vocabulary size({len(existing_vocabulary)}) "
+                f"of the original model. Some tokens will have to be dropped "
+                f"in order to continue training. It is advised to re-train the "
+                f"model from scratch on complete data."
             )
+        for token in vocabulary:
+            if token not in existing_vocabulary:
+                existing_vocabulary[token] = available_empty_index
+                del existing_vocabulary[f"{BUFFER_SLOTS_PREFIX}{available_empty_index}"]
+                available_empty_index += 1
+                if available_empty_index == len(existing_vocabulary):
+                    break
+        self._set_vocabulary(attribute, existing_vocabulary)
+
+    def _get_additional_vocabulary_size(
+        self, attribute: Text, existing_vocabulary_size: int
+    ) -> int:
+        """Get additional vocabulary size to be saved for incremental training.
+
+        If `self.additional_vocabulary_size` is not None,
+        we return that as the user should have specified
+        this number. If not then we take the default
+        additional vocabulary size which is 1/2 of the
+        current vocabulary size.
+        Args:
+            attribute: Message attribute for which additional vocabulary size should be computed.
+            existing_vocabulary_size: Current size of vocabulary learnt from the training data.
+
+        Returns:
+            Size of additional vocabulary that should be set aside for incremental training.
+        """
+        # Vocabulary expansion for INTENTS, ACTION_NAME
+        # and INTENT_RESPONSE_KEY is currently not supported.
+        if attribute not in DENSE_FEATURIZABLE_ATTRIBUTES:
+            return 0
+        if self.additional_vocabulary_size.get(attribute, None) is not None:
```

Comment:
```suggestion
        if self.additional_vocabulary_size.get(attribute) is not None:
```
`None` is the default return value of `get`